lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1646,6 +1646,14 @@ bool LLVMToSPIRV::transDecoration(Value *V, SPIRVValue *BV) {
           M |= FPFastMathModeNSZMask;
         if (FMF.allowReciprocal())
           M |= FPFastMathModeAllowRecipMask;
+        if (FMF.allowContract()) {
+          M |= FPFastMathModeAllowContractMask;
+          BM->addCapability(CapabilityFPFastMathModeINTEL);
```

Comment:
You should only do that if extension is enabled. Example: https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/625/files#diff-ba003da77a80df1db76fb378a873a80e758591d72bf82a46fc90069b2a947990R530